mcp/mcp_test.go

```diff
@@ -946,8 +946,8 @@ func TestAddTool_DuplicateNoPanicAndNoDuplicate(t *testing.T) {
 		// This case was written specifically to reproduce a bug where duplicate tools where causing jsonschema errors
 		t1 := &Tool{Name: "dup", Description: "first", InputSchema: &jsonschema.Schema{}}
 		t2 := &Tool{Name: "dup", Description: "second", InputSchema: &jsonschema.Schema{}}
-		s.AddTool(t1, nopHandler)
-		s.AddTool(t2, nopHandler)
+		s.AddRawTool(t1, nopHandler)
```

Comment:
This should just be AddTool. People won't understand what a 'RawTool' is, so let's not invent this distinction. We should instead explain in the documentation.